presto-base-arrow-flight/src/test/java/com/facebook/plugin/arrow/TestArrowFlightNativeQueries.java

```diff
@@ -0,0 +1,387 @@
+        assertQuery("SELECT CAST(CAST(linenumber as INTEGER) as DECIMAL(15, 6)) FROM lineitem");
+        assertQuery("SELECT CAST(nationkey as DECIMAL(18, 6)) FROM nation");
+
+        // Cast all integer types to long decimal
+        assertQuery("SELECT CAST(linenumber as DECIMAL(25, 0)) FROM lineitem");
+        assertQuery("SELECT CAST(linenumber as DECIMAL(19, 4)) FROM lineitem");
+        assertQuery("SELECT CAST(CAST(linenumber as INTEGER) as DECIMAL(20, 6)) FROM lineitem");
+        assertQuery("SELECT CAST(nationkey as DECIMAL(22, 6)) FROM nation");
+    }
+
+    @Test
+    public void testSwitch()
+    {
+        assertQuery("SELECT case linenumber % 10 when orderkey % 3 then orderkey + 1 when 2 then orderkey + 2 else 0 end FROM lineitem");
+        assertQuery("SELECT case linenumber when 1 then 'one' when 2 then 'two' else '...' end FROM lineitem");
+        assertQuery("SELECT case when linenumber = 1 then 'one' when linenumber = 2 then 'two' else '...' end FROM lineitem");
+    }
+
+    @Test
+    public void testIn()
+    {
+        assertQuery("SELECT linenumber IN (orderkey % 7, partkey % 5, suppkey % 3) FROM lineitem");
+    }
+
+    @Test
+    public void testSubqueries()
+    {
+        assertQuery("SELECT name FROM nation WHERE regionkey = (SELECT max(regionkey) FROM region)");
+
+        // Subquery returns zero rows.
+        assertQuery("SELECT name FROM nation WHERE regionkey = (SELECT regionkey FROM region WHERE regionkey < 0)");
+
+        // Subquery returns more than one row.
+        assertQueryFails("SELECT name FROM nation WHERE regionkey = (SELECT regionkey FROM region)", ".*Expected single row of input. Received 5 rows.*");
+    }
+
+    @Test
+    public void testArithmetic()
+    {
+        assertQuery("SELECT mod(orderkey, linenumber) FROM lineitem");
+        assertQuery("SELECT discount * 0.123 FROM lineitem");
+        assertQuery("SELECT ln(totalprice) FROM orders");
+        assertQuery("SELECT sqrt(totalprice) FROM orders");
+        assertQuery("SELECT radians(totalprice) FROM orders");
+    }
+
+    @Test
+    public void testGreatestLeast()
+    {
+        assertQuery("SELECT greatest(linenumber, suppkey, partkey) from lineitem");
+        assertQuery("SELECT least(shipdate, commitdate) from lineitem");
+    }
+
+    @Test
+    public void testSign()
+    {
+        assertQuery("SELECT sign(totalprice) from orders");
+        assertQuery("SELECT sign(-totalprice) from orders");
+        assertQuery("SELECT sign(custkey) from orders");
+        assertQuery("SELECT sign(-custkey) from orders");
+        assertQuery("SELECT sign(shippriority) from orders");
+    }
+
+    private static int findUnusedPort()
+            throws IOException
+    {
+        try (ServerSocket socket = new ServerSocket(0)) {
+            return socket.getLocalPort();
+        }
+    }
+
+    public static Map<String, String> getNativeWorkerSystemProperties()
```

Comment:
@BryanCutler : Can we use the functions from PrestoNativeQueryRunnerUtils for getting the properties and external worker launcher instead of repeating the code here ?

That would mean adding a dependency on presto-native-execution, but that should be fine right ?